packages/vhd-lib/openVhd.js

```diff
@@ -7,12 +7,12 @@ const { VhdFile } = require('./Vhd/VhdFile.js')
 exports.openVhd = async function openVhd(handler, path, opts) {
   const resolved = await resolveVhdAlias(handler, path)
   try {
-    return await VhdFile.open(handler, resolved, opts)
+    return await VhdDirectory.open(handler, resolved, opts)
   } catch (e) {
-    // if the remote is encrypted, trying to open a VhdFile will throw an assertion error before checking if the path is a directory, therefore we should try to open a VhdDirectory anyway.
-    if (e.code !== 'EISDIR' && e.code !== 'ERR_ASSERTION') {
+    // if the error is directory based , try to open as a vhd file
+    if (e.code !== 'EISDIR' && e.code !== 'ENOTDIR') {
```

Comment:
1. I find it difficult to understand, reversing the condition might help
2. I think it's weird that we fallback on VhdFile if the error is `EISDIR` or `ENOTDIR` :thinking: